### PR TITLE
DO NOT MERGE remove inputs/outputsToSave from runKernelFunc

### PR DESCRIPTION
**Pull request by tafsiri**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2947)

### CODE REVIEW

**Comment by tafsiri** [comment]
Fails on WASM. Once softmax is modularized, I'll look into modularing the other ops that wasm has kernels for so that we can remove inputsToSave/outputsToSave.
